res/app/control-panes/dashboard/environment/environment.pug

```diff
@@ -0,0 +1,35 @@
+.widget-container.fluid-height.stf-environment(ng-controller='EnvironmentCtrl')
+  .heading
+    stacked-icon(icon='fa-exchange', color='color-yellow')
+    span(translate) Switch enviroment
+    clear-button(ng-click='clear()', ng-disabled='!command && !data').btn-xs
+    help-icon(topic='Remote-Shell', uib-tooltip='{{"Switches Wandera\'s environment to DEV, QA, PROD or FI" | translate}}')
+  
+
+  .widget-content.padded
+    button.btn.btn-xs.btn-info-outline(ng-click='switchTo("DEV")',
+    uib-tooltip='{{ "Switch to DEV" | translate }}')
+      i.fa.fa-exchange
+      span(translate) DEV
+    button.btn.btn-xs.btn-info-outline(ng-click='switchTo("QA")',
+    uib-tooltip='{{ "Switch to QA" | translate }}')
+      i.fa.fa-exchange
+      span(translate) QA
+    button.btn.btn-xs.btn-info-outline(ng-click='switchTo("PRODUCTION")',
+    uib-tooltip='{{ "Switch to PRODUCTION" | translate }}')
+      i.fa.fa-exchange
+      span(translate) PRODUCTION
+    // NOTE: autofill doesn't work here
+    form(method='post', enable-autofill, ng-submit='run(command)')
+
+  .widget-content.padded
+    .input-group.form-inline
+      input(type=text, ng-model='command', Xtext-focus-select,
+      autocapitalize='off', spellcheck='false',
+      tabindex='30', accesskey='S', autocomplete='on' placeholder='custom environment url').form-control.environment-input 
```

Comment:
based on your implementation, it doesn't expect URL, but just FI name, so I would reword it
`'custom environment url'` -> `'FI name'` or something similar